api/app/Http/Controllers/EntityController.php

```diff
@@ -222,14 +237,149 @@ public function deleteOne($id)
     public function deleteMany(Request $request)
     {
         $requestCollection = $request->data;
-        $ids = $this->getIds($requestCollection, $this->getKeyName(), $this->validateRequestRule);
-        $models = $this->getRepository()->findMany($ids);
+        $models = $this->findOrFailCollection($requestCollection);
+        foreach ($models as $model) {
+            $model->delete();
+        }
+        return $this->getResponse()->noContent();
+    }
 
-        if (count($ids) !== $models->count()) {
-            throw $this->notFoundManyException($ids, $models, $this->getKeyName());
+    /**
+     * @param $id
+     * @return BaseModel
+     */
+    protected function findOrNewEntity($id)
+    {
+        $this->validateId($id, $this->getModel()->getKeyName(), $this->getIdValidationRule());
+
+        try {
+            return $this->getModel()->findByIdentifier($id);
+        } catch (ModelNotFoundException $e) {
+            return $this->getModel()->newInstance();
         }
+    }
 
-        $this->getRepository()->deleteMany($models);
-        return $this->getResponse()->noContent();
+    /**
+     * @param $id
+     * @return BaseModel
+     */
+    protected function findOrFailEntity($id)
+    {
+        $this->validateId($id, $this->getModel()->getKeyName(), $this->getIdValidationRule());
+
+        try {
+            return $this->getModel()->findByIdentifier($id);
+        } catch (ModelNotFoundException $e) {
+            throw $this->notFoundException($this->getModel()->getKeyName());
+        }
+    }
+
+    /**
+     * @return int
+     */
+    protected function countEntities()
+    {
+        return $this->getModel()->count();
+    }
+
+    /**
+     * @param null $limit
+     * @param null $offset
+     * @return Collection
+     */
+    protected function getAllEntities($limit = null, $offset = null)
+    {
+        return $this->getModel()->take($limit)->skip($offset)->get();
+    }
+
+    /**
+     * @param $requestCollection
+     * @return Collection
+     */
+    protected function findOrFailCollection($requestCollection)
+    {
+        $ids = $this->getIds($requestCollection, $this->getModel()->getKeyName(), $this->getIdValidationRule());
+
+        if (!empty($ids)) {
+            $models = $models = $this->getModel()->findMany($ids);
+        } else {
+            throw $this->notFoundManyException($ids, $this->getModel()->newCollection(), $this->getModel()->getKeyName());
```

Comment:
@Redjik you make an excellent point, sometimes a collection could be empty and it shouldn't be an exception.